src/Compilers/Helpers/CoreClrShim.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+
+namespace Roslyn.Utilities
+{
+
+    /// <summary>
+    /// This is a bridge for APIs that are only available on CoreCLR or .NET 4.6
+    /// and NOT on .NET 4.5. The compiler currently targets .NET 4.5 and CoreCLR
+    /// so this shim is necessary for switching on the dependent behavior.
+    /// </summary>
+    internal static class CoreClrShim
+    {
+        internal static bool IsCoreClr { get; } =
+            Type.GetType("System.Runtime.Loader.AssemblyLoadContext, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
```

Comment:
I tried that -- it didn't work. It looks like the contract assembly isn't copy-local according to the NuGet package resolution, so it isn't present at runtime.